### PR TITLE
[6.2.z] Refactor katello-agent related tests to use cdn/downstream repos accordingly

### PR DESCRIPTION
**Pull request by abalakh**
Basically reverts all the changes done with tests in #4439 #4449 #4462 #4463 and introduces another approach instead - helpers, not tests are updated to use the proper repo, this way tests require from little to no changes.
PR covers all places mentioned in #4429 so basically fully resolves it for 6.2.z branch
It also resolves issue #4153 as it's a blocker for downstream repo usage (assertions mentioned there are always failing for downstream repo, it was working for us only in cdn)

### CODE REVIEW

File: robottelo/cli/factory.py [hunk 1]
```diff
@@ -2549,12 +2590,13 @@ def _get_capsule_vm_distro_repos(distro):
             'arch': rh_product_arch
         })
         # Red Hat Satellite Capsule 6.2 (for RHEL 7 Server)
-        rh_repos.append({
-            'product': PRDS['rhsc'],
-            'repository-set': REPOSET['rhsc7'],
-            'repository': REPOS['rhsc7']['name'],
-            'repository-id': REPOS['rhsc7']['id'],
-        })
+        if settings.cdn:
```
Comment (abalakh) [inline robottelo/cli/factory.py]: @ldjebran these are the changes from #4552, they fit this PR much better.

**Comment by abalakh** [comment]
Now the hardest part - retest everything affected 😄  Will put DONOTMERGE label while gathering results.

**Comment by abalakh** [comment]
The rest of test results:
```python
py.test tests/foreman/cli/test_host.py -k test_positive_register_host_ak_with_host_collection
============================= test session starts ==============================
platform darwin -- Python 2.7.13, pytest-3.0.7, py-1.4.33, pluggy-0.4.0
rootdir: /Users/andrii/workspace/robottelo, inifile:
plugins: xdist-1.15.0, services-1.1.14, cov-2.3.1
collected 46 items
2017-04-11 15:33:04 - conftest - DEBUG - Found WONTFIX in decorated tests ['1156555', '1269196', '1402826', '1245334', '1221971', '1217635', '1226425', '1199150', '1204686', '1267224', '1103157', '1230902', '1214312', '1079482']

2017-04-11 15:33:04 - conftest - DEBUG - Collected 46 test cases


tests/foreman/cli/test_host.py .

============================= 45 tests deselected ==============================
================== 1 passed, 45 deselected in 432.40 seconds ===================

py.test tests/foreman/ui/test_dashboard.py -k test_positive_content_host_subscription_status
============================= test session starts ==============================
platform darwin -- Python 2.7.13, pytest-3.0.7, py-1.4.33, pluggy-0.4.0
rootdir: /Users/andrii/workspace/robottelo, inifile:
plugins: xdist-1.15.0, services-1.1.14, cov-2.3.1
collected 27 items
2017-04-12 12:57:05 - conftest - DEBUG - Found WONTFIX in decorated tests ['1156555', '1269196', '1402826', '1245334', '1221971', '1217635', '1226425', '1199150', '1204686', '1267224', '1103157', '1230902', '1214312', '1079482']

2017-04-12 12:57:05 - conftest - DEBUG - Collected 27 test cases


tests/foreman/ui/test_dashboard.py .

============================= 26 tests deselected ==============================
================== 1 passed, 26 deselected in 323.48 seconds ===================
```

File: robottelo/cli/factory.py [hunk 2]
```diff
@@ -2468,6 +2471,44 @@ def setup_org_for_a_rh_repo(options=None):
...
+    :param options: a dict with options to pass to function
+        ``_setup_org_for_a_rh_repo``. See its docstring for more details
+    :param force_manifest_upload: bool flag whether to upload a manifest to
+        organization even if downstream custom repo is used instead of CDN.
+        Useful when test relies on organization with manifest (e.g. uses some
+        other RH repo afterwards). Defaults to False.
+    :return: a dict with entity ids (see ``_setup_org_for_a_rh_repo`` and
+        ``setup_org_for_a_custom_repo``).
+    """
+    repo_url = None
```
Comment (ldjebran) [inline robottelo/cli/factory.py]: to be more explicit, we can rename it to custom_repo_url

File: robottelo/cli/factory.py [hunk 3]
```diff
@@ -2468,6 +2471,44 @@ def setup_org_for_a_rh_repo(options=None):
     }
 
 
+def setup_org_for_a_rh_repo(options=None, force_manifest_upload=False):
```
Comment (ldjebran) [inline robottelo/cli/factory.py]: we can add a kwarg force_use_cdn=False for the tests that need only cdn like "test_positive_add_redhat_product"
that will call this function with force_use_cdn=True instead of using private function

File: tests/foreman/cli/test_activationkey.py [hunk 4]
```diff
@@ -620,7 +620,9 @@ def test_positive_add_redhat_product(self):
         @CaseLevel: System
         """
         org = make_org()
-        result = setup_org_for_a_rh_repo({
+        # exceptional case here. we need this repo to be RH one no matter are
+        # we in downstream or cdn
+        result = _setup_org_for_a_rh_repo({
```
Comment (ldjebran) [inline tests/foreman/cli/test_activationkey.py]: look at my comment related to force_use_cdn=True

File: robottelo/vm.py [hunk 5]
```diff
@@ -249,6 +260,9 @@ def install_katello_agent(self):
             raise VirtualMachineError('Failed to install katello-agent')
         gofer_check = self.run('service goferd status')
         if gofer_check.return_code != 0:
+            self.run('service goferd start')
```
Comment (abalakh) [inline robottelo/vm.py]: I hope there's no need anymore - bz is finally on_qa, mentioned updated packages are already available downstream. Will try to revert this change and see how it goes.

**Comment by abalakh** [comment]
Comments addressed. Test results for affected tests (with workaround to manually start katello-agent):
```python
py.test tests/foreman/ui/test_dashboard.py -k test_positive_content_host_subscription_status
============================= test session starts ==============================
platform darwin -- Python 2.7.13, pytest-3.0.7, py-1.4.33, pluggy-0.4.0
rootdir: /Users/andrii/workspace/robottelo, inifile:
plugins: xdist-1.15.0, services-1.1.14, cov-2.3.1
collected 27 items
2017-04-13 17:40:39 - conftest - DEBUG - Found WONTFIX in decorated tests ['1156555', '1269196', '1402826', '1245334', '1221971', '1217635', '1226425', '1199150', '1204686', '1267224', '1103157', '1230902', '1214312', '1079482']

2017-04-13 17:40:39 - conftest - DEBUG - Collected 27 test cases


tests/foreman/ui/test_dashboard.py .

============================= 26 tests deselected ==============================
================== 1 passed, 26 deselected in 459.02 seconds ===================

py.test tests/foreman/cli/test_activationkey.py -k 'test_positive_add_redhat_product or test_positive_add_redhat_and_custom_products'
============================= test session starts ==============================
platform darwin -- Python 2.7.13, pytest-3.0.7, py-1.4.33, pluggy-0.4.0
rootdir: /Users/andrii/workspace/robottelo, inifile:
plugins: xdist-1.15.0, services-1.1.14, cov-2.3.1
collected 48 items
2017-04-13 16:31:11 - conftest - DEBUG - Found WONTFIX in decorated tests ['1156555', '1269196', '1402826', '1245334', '1221971', '1217635', '1226425', '1199150', '1204686', '1267224', '1103157', '1230902', '1214312', '1079482']

2017-04-13 16:31:11 - conftest - DEBUG - Collected 48 test cases


tests/foreman/cli/test_activationkey.py ..

============================= 46 tests deselected ==============================
================== 2 passed, 46 deselected in 647.81 seconds ===================
```

**Comment by abalakh** [comment]
@lpramuk PR for master is submitted (#4572), it's not a cherry-pick though, as too many things differ and it was easier and safer to analyze each affected test from the very beginning.

File: tests/foreman/ui/test_dashboard.py [hunk 6]
```diff
@@ -638,13 +640,12 @@ def test_positive_content_host_subscription_status(self):
             'content-view-id': content_view.id,
             'lifecycle-environment-id': env.id,
             'activationkey-id': activation_key.id,
-        })
+        }, force_use_cdn=True)
```
Comment (rplevka) [inline tests/foreman/ui/test_dashboard.py]: well, it actually causes the test to fail as cdn version won't start the agent up

Comment (abalakh) [inline tests/foreman/ui/test_dashboard.py]: True. There's not much we can do about it - either introduce a workaround (which you wanted to) or wait till next cdn build.
